src/content/docs/en/reference/dev-toolbar-app-reference.mdx

```diff
@@ -3,277 +3,369 @@ title: Dev Toolbar App API
+### On the client
 
-```ts title="src/my-app.js" {4-10}
-  export default {
-    // ...
-    init(canvas, eventTarget) {
-      eventTarget.dispatchEvent(
-        new CustomEvent('toggle-app', {
-          detail: {
-            state: false,
-          },
-        })
-      );
-    }
-  }
+In your app, use the [`server` object on the `init()` hook](#server) to send and receive messages to and from the server.
+
+```ts title="src/my-app.js" {3-7} "server"
+export default defineToolbarApp({
+  init(canvas, app, server) {
+    server.send('my-message', { message: 'Hello!' });
+
+    server.on('server-message', (data) => {
+      console.log(data.message);
+    });
+  },
+});
 ```
 
-#### `state: boolean`
+#### `send()`
 
-Indicates whether or not the app should be enabled. When `true`, the app will be enabled. Conversely, when `false`, the app will be disabled. If this property is not specified, `true` will be assumed.
+**Signature:** `send<T>(event: stringify, data: T) => void`
 
-## Client-Server Communication
+Sends data to the server from logic defined in your toolbar app.
+
+```ts title="src/my-app.js"
+init(canvas, app, server) {
+  server.send('my-app:my-message', { message: 'Hello!' });
+}
+```
 
-Using [Vite's methods for client-server communication](https://vitejs.dev/guide/api-plugin.html#client-server-communication), Dev Toolbar Apps can communicate with the server.
+When sending messages from the client to the server, it is good practice to prefix the event name with the app ID or other namespaces to avoid conflicts with other apps or other integrations that may be listening for messages.
 
-In addition to being able to send and receive custom messages, the Dev Toolbar also sends the following messages, where `APP_ID` is the [app's ID](#id-string):
+#### `on()`
 
-### `astro-dev-toolbar:APP_ID:initialized`
+**Signature:** `on<T>(event: string, callback: (data: T) => void) => void`
 
-This message is sent when the app is initialized. The data for this message is empty.
+Registers a callback to be called when the server sends a message with the specified event.
 
-```ts title="integration.ts" {4-6}
-{
-  // ...
-  "astro:server:setup": ({ server }) => {
-    server.ws.on("astro-dev-toolbar:super-app:initialized", () => {
-      console.log("My app was initialized!");
-    });
-  },
-  // ...
+```ts title="src/my-app.js"
+init(canvas, app, server) {
+  server.on('server-message', (data) => {
+    console.log(data.message);
+  });
 }
 ```
 
-### `astro-dev-toolbar:APP_ID:toggled`
+### On the server
 
-This message is sent when the user clicks on the app icon in the UI. The data for this message is a boolean indicating whether the app is enabled or not.
+In [the integration that adds your toolbar app](#toolbar-app-integration-setup), use the [`astro:server:setup` hook](/en/reference/integrations-reference/#astroserversetup) to access the `toolbar` object to send and receive messages to and from your apps.
```

Comment:
It can be in any integration, not just the one where the toolbar app is added. This is not super important, but it's a neat thing to be able to do cross integration/app communication